segmentation/utils/visualize.py

```diff
@@ -1,20 +1,152 @@
+                  population_col_name: Name of column in dataframe containing categorical Population data
+                  color_map: Name of MatPlotLib ColorMap used, default is jet
+                  show_total_count: Boolean specifying whether to show graph of total population count, default is true
+                  show_distribution: Boolean specifying whether to show graph of population distribution, default is true
+                  show_proportion: Boolean specifying whether to show graph of total count, default is true
+                  save_dir: Directory to save plots, default is blank"""
+    cell_data = cell_data.dropna()
+
+    # Plot by total count
+    if show_total_count:
+        population_values = cell_data[population_col_name].value_counts()
+        population_values.plot.bar(colormap=color_map)
+        plt.title("Distribution of Population in all patients")
+        plt.xlabel("Population Type")
+        plt.ylabel("Population Count")
+        plt.savefig(save_dir + "TotalPopulationDistribution.png")
+
+    # Plot by count
+    if show_distribution:
+        getSortedData(cell_data, patient_col_name, population_col_name).plot.bar(stacked=True, colormap=color_map)
+        plt.legend(loc='center left', bbox_to_anchor=(1.0, 0.5))
+        plt.xlabel("Patient ID")
+        plt.ylabel("Cell Count")
+        plt.title("Distribution of Population Count in Patients")
+        plt.savefig(save_dir + "PopulationDistribution.png")
+
+    # Plot by Proportion
+    if show_proportion:
+        getSortedData(cell_data, patient_col_name, population_col_name, is_normalized=True).plot.bar(stacked=True,
+                                                                                                     legend=False,
+                                                                                                     colormap=color_map)
+        plt.legend(loc='center left', bbox_to_anchor=(1.0, 0.5))
+        plt.xlabel("Patient ID")
+        plt.ylabel("Population Proportion")
+        plt.title("Distribution of Population Count Proportion in Patients")
+        plt.savefig(save_dir + "PopulationProportion.png")
+
+
+def visualize_dimensionality_reduction(cell_data, columns, category, color_map="Spectral", algorithm="UMAP", save_dir=""):
+    """Plots the dimensionality reduction of specified population columns
+                Args:
+                    cell_data: Pandas Dataframe containing columns for dimensionality reduction and category
+                    columns: List of column names that are included for dimensionality reduction
+                    category: Name of column in dataframe containing categorical Population or Patient data
+                    color_map: Name of MatPlotLib ColorMap used, default is Spectral
+                    algorithm: Name of dimensionality reduction algorithm, default is UMAP
+                    save_dir: Directory to save plots, default is blank"""
```

Comment:
Make sure to include the type of each argument in parentheses, and also to include the return value and type.